### PR TITLE
DO_NOT_MERGE: prototype of IPD/HMD movement scaling

### PR DESCRIPTION
**Pull request by hyperlogic**
- There is a new userHeight parameter in preferences (also exposed to script)
- There is a new MyAvatar getAvatarHeight() method that returns the height of the currently worn avatar, including avatar scale factor.
- The IPD and HMD and Hand controller motion is scaled by avatarHeight / userHeight.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 092f25425bf338946f95173b1bd9d15727a2369c available at: https://gist.github.com/65c03a3501829eb65b3bf2337079ba9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-092f254.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-092f254.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c6432ab4674c15c0ec5b8b4a42d37124074d9e5 available at: https://gist.github.com/65c03a3501829eb65b3bf2337079ba9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-8c6432a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-8c6432a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4ed1c4da8ffb783d91eb727c712d96427c27f04f available at: https://gist.github.com/65c03a3501829eb65b3bf2337079ba9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-4ed1c4d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-4ed1c4d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 424c30cdbcdbcb4c8061498f4ca202c89aba47ea available at: https://gist.github.com/65c03a3501829eb65b3bf2337079ba9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-424c30c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-424c30c.dmg

**Comment by thoys** [comment]
hi @hyperlogic , can you please update this with the latest master?

**Comment by hifi-gustavo** [comment]
Build console output for commit: d63cedb8c6b3485936dd50102e90cc175b21369f available at: https://gist.github.com/65c03a3501829eb65b3bf2337079ba9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-d63cedb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8694-d63cedb.dmg

File: interface/src/avatar/Avatar.cpp [hunk 1]
```diff
@@ -1238,6 +1238,27 @@ glm::quat Avatar::getUncachedRightPalmRotation() const {
     return rightPalmRotation;
 }
 
+// OUTOFBODY_HACK:  move these constants to a common place.
+static const float DEFAULT_AVATAR_EYE_HEIGHT = 1.75f;
+static const float EYE_TO_TOP_OF_HEAD = 0.11f;  // 11 cm  https://en.wikipedia.org/wiki/Human_head#/media/File:HeadAnthropometry.JPG
```
Comment (thoys) [inline interface/src/avatar/Avatar.cpp]: Who says all avatars will be human? ;)

Comment (thoys) [inline interface/src/avatar/Avatar.cpp]: You could calculate 'EyeHeight' to 'HeadTop_End' or 'HeadEnd', if any of these joints exist

**Comment by ZappoMan** [comment]
@hyperlogic - I'm closing this PR for now, since it's almost a month old and has a merge conflict... please revisit it in the context of future work.

**Pull request closed without merging**